polkadot/node/subsystem-types/src/lib.rs

```diff
@@ -22,10 +22,19 @@
 
 #![warn(missing_docs)]
 
+use smallvec::SmallVec;
 use std::{fmt, sync::Arc};
 
-pub use polkadot_primitives::{BlockNumber, Hash};
-use smallvec::SmallVec;
+pub use polkadot_primitives::{Block, BlockNumber, Hash};
+
+/// Keeps the state of a specific block pinned in memory while the handle is alive.
+///
+/// The handle is reference counted and once the last is dropped, the
+/// block is unpinned.
+///
+/// This is useful for runtime API calls to blocks that are
+/// racing against finality, e.g. for slashing purposes.
+pub type UnpinHandle = sc_client_api::UnpinHandle<Block>;
```

Comment:
This is awesome! No races against finality anymore! This should do away with a couple of issues, you have been looking into @tdimitrov !

Reply:
All we need to do is keep the leaf alive, while we work on it.